numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -317,18 +317,31 @@ PyArray_ConvertMultiAxis(PyObject *axis_in, int ndim, npy_bool *out_axis_flags)
 NPY_NO_EXPORT int
 PyArray_BoolConverter(PyObject *object, npy_bool *val)
 {
-    if (PyObject_IsTrue(object)) {
+    /* Super fast path, for the most common input: */
```

Comment:
This seems pointless, it's already the first line of `PyObject_IsTrue`

Reply:
Hehe, yeah, I was trying aroudn a bit too much, it probably shaved of a nanosecond or two somewhere, which is just not necessary.